firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java

```diff
@@ -689,4 +691,76 @@ public void testCanonicalIdsAreStable() {
   private void assertCanonicalId(Query query, String expectedCanonicalId) {
     assertEquals(expectedCanonicalId, query.toTarget().getCanonicalId());
   }
+
+  @Test
+  public void testOrQuery() {
+    MutableDocument doc1 = doc("collection/1", 0, map("a", 1, "b", 0));
+    MutableDocument doc2 = doc("collection/2", 0, map("a", 2, "b", 1));
+    MutableDocument doc3 = doc("collection/3", 0, map("a", 3, "b", 2));
+    MutableDocument doc4 = doc("collection/4", 0, map("a", 1, "b", 3));
+    MutableDocument doc5 = doc("collection/5", 0, map("a", 1, "b", 1));
+
+    // Two equalities: a==1 || b==1.
+    Query query1 =
+        Query.atPath(ResourcePath.fromString("collection"))
+            .filter(orFilters(filter("a", "==", 1), filter("b", "==", 1)));
+    assertQueryMatches(
+        query1,
+        /* match */ Arrays.asList(doc1, doc2, doc4, doc5),
+        /* not match */ Arrays.asList(doc3));
```

Comment:
Unrelated thought: What should the canonical ID of a DNF-query and an equivalent non-DNF query be? Should they match? Should we create one or two listens?

Reply:
For the time being, the canonical ID of `a || b` is not the same as `b || a`. We can discuss this further, and address it in the next PR.